experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -236,9 +273,13 @@ func VerifyRekorEntry(rekorLogEntryBytes []byte, productTeamKeyBytes []byte, end
 	}
 
 	// Check that hash of endorsement file matches hash of rekor log entry contents.
-	err = compareEndorsementAndRekorHash(rekordEntry, endorsementFileBytes)
+	err = compareEndorsementAndRekorHash(rekordEntry, endorsementBytes)
```

Comment:
pre-existing: I think this function assumes that the input endorsement file is in canonicalized json format. But it might not always be the case. In fact, I think it is more likely to receive the endorsement file in a pretty format. The code should first convert the endorsement file into the canicalized format.

Reply:
Or maybe not? I am not sure if the artefact needs to be canonicalized.

Reply:
Hmmm... I think it *should not be canonicalized* (which might be strange, but here's why I think this):

It looks like the implementation of `sha256.Sum256` does not do any canonicalizing because it just takes raw bytes.

However, because this test is passing, it looks like when a rekor entry is created, rekor does not do any canonicalizing (which makes sense because if I understand correctly, any file can be used as a rekor entry). So to make sure the hashes are equal, the endorsement files should just be in the same format (canonicalized or not) both at the time the rekor entry is created and at the time the bytes are supplied to this wrapper.

Reply:
Does that seem OK? WDYT?

Reply:
I agree. It does need to be canonicalized. I also managed to generate the same hash in the Rust code, without cononicalizing the endorsement file.